### DISCUSSION TITLE
gatsby-plugin-image StaticImage not loading on Gatsby Cloud deployment

### DISCUSSION THREAD

**Discussion by totallyspun** [gatsbyjs/gatsby#36045] [Help]
Hi, I have recently completed the tutorial, but still have one nagging issue that I have not be able to resolve. I'm not sure if this is actually a bug in the cloud or something I've set up wrong. I didn't add this to the similar discussion existing because that was not deploying via Gatsby Cloud.

The image `opossom.jpg` is in the relative path shown below. I can confirm this is correct because the image shows in my browser when I do `gatsby develop` OR `gatsby build && gatsby serve`. The later steps of the tutorial with GatsbyImage components also work fine both locally and deployed. However, when I push to my repository and trigger a build and deployment in the cloud, I get the following error related to the StaticImage component.

```
static-image.server.tsx:48 No data found for image "../images/opossom.jpg"
              undefinedCould not read image data file "/usr/src/app/www/.cache/caches/gatsby-plugin-image/826667929.json". 
This may mean that the images in "/usr/src/app/www/src/pages/index.js" were not processed.
Please ensure that your gatsby version is at least 2.24.78.
```

I am running Gatsby 4.17.2 so the final line is irrelevant. (Note: I had to downgrade React and React DOM at a later step in the tutorial because of a peer dependency error when installing `gatsby-plugin-mdx`; however, I can confirm this error was happening even when all packages were shiny)

Build deployed at https://https://paulrgatsbymain.gatsbyjs.io/
Repository at https://github.com/totallyspun/paulr-gatsby
Node version: 14.19.3

Here is my index.js
```
import * as React from "react";
import Layout from "../components/layout";
import { StaticImage } from "gatsby-plugin-image";

const IndexPage = () => {
  return (
    <Layout pageTitle="Home Page">
      <p>What the hell?</p>
      <StaticImage
        src="../images/opossom.jpg"
        alt="an opossum under a canoe"
        placeholder="blurred"
      />
    </Layout>
  )
}

export default IndexPage
```

Thanks in advance for any assistance

**Comment by jongwooo** [comment]
Add this setting to `gatsby-config.js`.
```js
// gatsby-config.js

module.exports = {
  plugins: [
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `images`,
        path: `${__dirname}/src/images`,
      },
    },
  ],
}
```

**Comment by totallyspun** [comment]
Hi, thanks for the quick reply!

Unfortunately this makes no difference whatsoever.

Here's the build with the update you suggested. It loads but without image, same errors in console.
https://build-dd240e07-9513-42db-b633-998229974866.gtsb.io/

I then also did a 'clear cache and trigger build' action and now just get the build logs when I try to load the main build url

**Reply by jongwooo** [reply]
@totallyspun Capitalize the image file extension in the props of the component. I hope this helps.

```js hl_lines="1"
import * as React from "react";
import Layout from "../components/layout";
import { StaticImage } from "gatsby-plugin-image";

const IndexPage = () => {
  return (
    <Layout pageTitle="Home Page">
      <p>What the hell?</p>
      <StaticImage
        src="../images/opossom.JPG"
        alt="an opossum under a canoe"
        placeholder="blurred"
      />
    </Layout>
  )
}

export default IndexPage
```

**Comment by totallyspun** [comment]
So this is infuriating.
When I first encountered this error I did change the extension on disk to lower case
But I just learned that this change doesn't get seen by Git without explicitly calling `git mv ...` so I've done that and cleared the cache in Gatsby Cloud and finally it works!


![Screen Shot 2022-07-02 at 1 20 59 PM](https://user-images.githubusercontent.com/80244/177010774-5abe90f5-fd8a-4470-a76c-e4aee4911d2c.png)

Thanks @jongwooo for having an Eagle eye to see that. I was never going to find it looking at local files!
